synapse/http/federation/matrix_federation_agent.py

```diff
@@ -190,6 +240,45 @@ def _route_matrix_uri(self, parsed_uri):
             target_port=port,
         ))
 
+    @defer.inlineCallbacks
+    def get_well_known(self, server_name):
+        # FIXME: add a cache
+
+        uri = b"https://%s/.well-known/matrix/server" % (server_name, )
+        logger.debug("Fetching %s", uri.decode("ascii"))
+        try:
+            response = yield make_deferred_yieldable(
+                self._well_known_agent.request(b"GET", uri),
+            )
+        except Exception as e:
+            logger.info(
+                "Connection error fetching %s: %s",
+                uri.decode("ascii"), e,
+            )
+            defer.returnValue(None)
+
+        body = yield make_deferred_yieldable(readBody(response))
+
+        logger.info("Response from .well-known: %s", body)
```

Comment:
This should probably be `debug`

Reply:
I'm minded to leave it as info for now as otherwise it's quite hard to see what's going on. I've moved it down a bit to print out the parsed response though.